ee/clickhouse/queries/test/test_paths.py

```diff
@@ -15,15 +18,104 @@ def _create_event(**kwargs):
     create_event(**kwargs)
 
 
-class TestClickhousePaths(ClickhouseTestMixin, paths_test_factory(ClickhousePaths, _create_event, Person.objects.create)):  # type: ignore
+class TestClickhousePathsOld(ClickhouseTestMixin, paths_test_factory(ClickhousePaths, _create_event, Person.objects.create)):  # type: ignore
+    # remove when migrated to new Paths query
+    def test_denormalized_properties(self):
+        materialize("events", "$current_url")
+        materialize("events", "$screen_name")
+
+        query, _ = ClickhousePathsNew(team=self.team, filter=PathFilter(data={"path_type": PAGEVIEW_EVENT})).get_query()
+        self.assertNotIn("json", query.lower())
+
+        query, _ = ClickhousePathsNew(team=self.team, filter=PathFilter(data={"path_type": SCREEN_EVENT})).get_query()
+        self.assertNotIn("json", query.lower())
+
+        self.test_current_url_paths_and_logic()
+
+
+class TestClickhousePaths(ClickhouseTestMixin, paths_test_factory(ClickhousePathsNew, _create_event, Person.objects.create)):  # type: ignore
     def test_denormalized_properties(self):
         materialize("events", "$current_url")
         materialize("events", "$screen_name")
 
-        query, _ = ClickhousePaths().get_query(team=self.team, filter=PathFilter(data={"path_type": PAGEVIEW_EVENT}))
+        query, _ = ClickhousePathsNew(team=self.team, filter=PathFilter(data={"path_type": PAGEVIEW_EVENT})).get_query()
         self.assertNotIn("json", query.lower())
 
-        query, _ = ClickhousePaths().get_query(team=self.team, filter=PathFilter(data={"path_type": SCREEN_EVENT}))
+        query, _ = ClickhousePathsNew(team=self.team, filter=PathFilter(data={"path_type": SCREEN_EVENT})).get_query()
         self.assertNotIn("json", query.lower())
 
         self.test_current_url_paths_and_logic()
+
+    def test_paths_start(self):
```

Comment:
this is pretty suspect, trying to figure out why we have the off by one error in the new query. Judging from how the data is setup, 

```
{"source": "1_/pricing", "target": "2_/about", "value": 1}
```

is incorrect, the value should be 2, like in the postgres test.

Reply:
@paolodamico @kpthatsme @marcushyett-ph When we talk about setting a start point for paths, should the start point define the beginning of the session that a person is in or should it include paths after the start point even if it's in the middle of the session? For example, if one user has path A -> B -> C and another has B -> C and out start point is B. Would we count both of them or just the second person because the second person starts at B

Reply:
If I'm reading your question right–

I think the path we show should start from whichever event the user is saying is their starting point.

For example, I may want to know everything that happens after a user ingests data:

So the path should only show the journey starting with the "user ingested data" event.

In your example case, I think we'd include both of these users - both starting from B onwards

Reply:
Yeah makes sense. We wanted to make sure there's no a significant need/possible confusion for the other method where start points are defined as exact starts for a person session. I think the use case you described is much more useful though and it's how our current paths functionality works so we'll stick with that